lib/client.ts

```diff
@@ -88,35 +76,49 @@ export class Client {
     this.authorize(this.request(this.url(endpoint), { method: 'GET' }, reqConfig))
 
   public post = (endpoint: Endpoint, payload: any, reqConfig?: RequestConfig): Promise<any> =>
-    this.authorize(this.request(this.url(endpoint), { method: 'POST' , body: JSON.stringify(payload)}, reqConfig))
+    this.authorize(this.request(this.url(endpoint), { method: 'POST', body: JSON.stringify(payload) }, reqConfig))
 
   public patch = (endpoint: Endpoint, payload: any, reqConfig?: RequestConfig): Promise<any> =>
-    this.authorize(this.request(this.url(endpoint), { method: 'PATCH', body: JSON.stringify(payload)}, reqConfig))
+    this.authorize(this.request(this.url(endpoint), { method: 'PATCH', body: JSON.stringify(payload) }, reqConfig))
 
   public destroy = (endpoint: Endpoint, payload?: any, reqConfig?: RequestConfig): Promise<any> =>
-    this.authorize(this.request(this.url(endpoint), { method: 'DELETE', body: JSON.stringify(payload)}, reqConfig))
-
-  private url = (endpoint: Endpoint): string => ifElse(
-    is(String),
-    concat(this.host),
-    this.urlConfig
-  )(endpoint)
-
-  private urlConfig = ({ base, action, params = {}, qs }: EndpointConfig): string => compose(
-    when(
-      () => notNil(qs),
-      finalUrl => `${finalUrl}?${stringify(qs, { arrayFormat: 'brackets' })}`
-    ),
-    when(
-      () => notNil(action),
-      resourceUrl => `${resourceUrl}/${action}`
-    ),
-    when(
-      () => notNil(params.id),
-      collectionUrl => `${collectionUrl}/${params.id}`
-    ),
-    (hostname) => `${hostname}${S(base).template(params, '{', '}').s}`
-  )(this.host)
+    this.authorize(this.request(this.url(endpoint), { method: 'DELETE', body: JSON.stringify(payload) }, reqConfig))
+
+  private url = (endpoint: Endpoint): string =>
+    notNil(endpoint) && (endpoint.constructor === String || endpoint instanceof String) ?
+      `${this.host}${endpoint}` : // TODO: Do we want to allow this. Version will not be handled for this
```

Comment:
This is probably one of the main questions for code review.

My vision: JS-SDK **should not** support the `endpoint instanceof String` login anymore